### PR TITLE
Fix automatic reconnection in MQTT transport

### PR DESCRIPTION
**Pull request by feymartynov**
Paho calls `onSuccess` callback of `MQTTAsync_connect` call only on the first connection attempt.
In case of reconnection due to network problems or broker restart it wouldn't be called hence when back online the transport would not notify event handlers, send status message and resubscribe to the topic (which is necessary when `cleansession` flag is enabled in config).
Instead of `onSuccess` we [should](https://www.eclipse.org/lists/paho-dev/msg04046.html) set another callback with `MQTTAsync_setConnected` which is being called on reconnection and do all the above stuff there.

### CODE REVIEW

**Comment by lminiero** [comment]
Thanks! I'm currently on vacation, I'll have a look when I'm back at the office.

**Comment by lminiero** [comment]
Apologies, I've just returned from my vacation. I'll have a look either today or on Monday. Thanks!

File: transports/janus_mqtt.c [hunk 1]
```diff
@@ -720,48 +780,13 @@ int janus_mqtt_client_connect(janus_mqtt_context *ctx) {
 }
 
 void janus_mqtt_client_connect_success(void *context, MQTTAsync_successData *response) {
-	janus_mqtt_client_connect_success_impl(context);
```
Comment (lminiero) [inline transports/janus_mqtt.c]: Please don't keep methods empty. At the very least, there should be a comment explaining why we don't care about this callback.

File: transports/janus_mqtt.c [hunk 2]
```diff
@@ -720,48 +780,13 @@ int janus_mqtt_client_connect(janus_mqtt_context *ctx) {
...
 }
 
 #ifdef MQTTVERSION_5
 void janus_mqtt_client_connect_success5(void *context, MQTTAsync_successData5 *response) {
-	janus_mqtt_client_connect_success_impl(context);
```
Comment (lminiero) [inline transports/janus_mqtt.c]: See above.

**Comment by lminiero** [comment]
@feymartynov please ping me as soon as you fix those couple of nits above, as I think that apart from that this is ready to be merged. Thanks!

**Comment by feymartynov** [comment]
@lminiero updated the PR.

**Comment by lminiero** [comment]
Thanks, merging! :+1:

**Pull request merged**